src/main/java/refinedstorage/tile/externalstorage/ItemStorageDrawer.java

```diff
@@ -41,47 +41,31 @@ private boolean isVoidable() {
     @Override
     public ItemStack insertItem(ItemStack stack, int size, boolean simulate) {
         if (IFilterable.canTake(externalStorage.getItemFilters(), externalStorage.getMode(), externalStorage.getCompare(), stack) && drawer.canItemBeStored(stack)) {
-            if (!drawer.isEmpty()) {
-                if (getStored() + size > drawer.getMaxCapacity(stack)) {
-                    int remainingSpace = getCapacity() - getStored();
+            int stored = drawer.getStoredItemCount();
+            int remainingSpace = drawer.getMaxCapacity(stack) - stored;
 
-                    if (remainingSpace <= 0) {
-                        return isVoidable() ? null : ItemHandlerHelper.copyStackWithSize(stack, size);
-                    }
+            int inserted = remainingSpace > size ? size : (remainingSpace <= 0) ? 0 : remainingSpace;
 
-                    if (!simulate) {
-                        drawer.setStoredItemCount(drawer.getStoredItemCount() + remainingSpace);
-                    }
-
-                    return isVoidable() ? null : ItemHandlerHelper.copyStackWithSize(stack, size - remainingSpace);
-                } else {
-                    if (!simulate) {
-                        drawer.setStoredItemCount(drawer.getStoredItemCount() + size);
-                    }
-
-                    return null;
+            if (!simulate && remainingSpace > 0) {
+                if (drawer.isEmpty()) {
+                    drawer.setStoredItemRedir(stack, inserted);
                 }
-            } else {
-                if (getStored() + size > drawer.getMaxCapacity(stack)) {
-                    int remainingSpace = getCapacity() - getStored();
-
-                    if (remainingSpace <= 0) {
-                        return isVoidable() ? null : ItemHandlerHelper.copyStackWithSize(stack, size);
-                    }
+                else {
```

Comment:
It's not. Look at the file in your repo: https://github.com/geldorn/refinedstorage/blob/9c38c2998333cb4ab9a33d973ce29aa487bdfb9d/src/main/java/refinedstorage/tile/externalstorage/ItemStorageDrawer.java#L53-L55